python/graphstorm/model/rgcn_encoder.py

```diff
@@ -155,13 +154,17 @@ def forward(self, g, inputs):
 
         if g.is_block:
             inputs_src = inputs
-            inputs_dst = {k: v[:g.number_of_dst_nodes(k)] for k, v in inputs.items()}
+            if self.self_loop:
+                inputs_dst = {k: v[:g.number_of_dst_nodes(k)] for k, v in inputs.items()}
+            else:
+                # If we use RGCN for constructing node features, the destination nodes
+                # may not in the input features.
+                inputs_dst = {k: th.zeros((g.num_dst_nodes(k), self.in_feat),
```

Comment:
If a user initializes RelGraphConvLayer with self_loop set to false, does self_loop = False share the same implementation for constructing node features?

Reply:
in this case, the dst embeddings are not used. does it matter?